### PR TITLE
[Parent][MBL-13618] Announcement Details

### PR DESCRIPTION
**Pull request by Blackchai**
Couple of things to note:
-Institution level announcements that don't target the observer role will result in an error screen. This behavior is consistent with the current playstore version, I'm not sure if this is an api issue or not. It seems odd that the alert would show up, but not be viewable (api returns a 401).
-The webview in use needs a bit of work. I'm going to start on another commit that adds an extension function to format/load Html, so we can use the html_wrapper consistently. We also need to handle a few of the things that CanvasWebView is doing for us, so things like the new RCE images work.

### CODE REVIEW

File: apps/flutter_parent/lib/models/alert.dart [hunk 1]
```diff
@@ -95,6 +95,15 @@ abstract class Alert implements Built<Alert, AlertBuilder> {
       AlertType.courseGradeLow,
     ].contains(alertType);
   }
+
+  String getCourseId() {
```
Comment (Blackchai) [inline apps/flutter_parent/lib/models/alert.dart]: Just remembered this little fella. I need to write a unit test or two for it.

**Comment by instructure-cx** [comment]
Scan submitted to Checkmarx

**Comment by instructure-cx** [comment]
Scan submitted to Checkmarx

**Comment by instructure-cx** [comment]
Scan submitted to Checkmarx

**Comment by instructure-cx** [comment]
### Checkmarx scan completed
[Full Scan Details](https://cxmgr.inssec.info/CxWebClient/ViewerMain.aspx?scanid=1018962&projectid=20)
#### Checkmarx Scan Summary
Severity|Count
---|---
High|0
Medium|0
Low|0
Informational|0

#### Violation Summary
Severity|Count
---|---

**Comment by instructure-cx** [comment]
### Checkmarx scan completed
[Full Scan Details](https://cxmgr.inssec.info/CxWebClient/ViewerMain.aspx?scanid=1018963&projectid=20)
#### Checkmarx Scan Summary
Severity|Count
---|---
High|0
Medium|0
Low|0
Informational|0

#### Violation Summary
Severity|Count
---|---

**Comment by instructure-cx** [comment]
### Checkmarx scan completed
[Full Scan Details](https://cxmgr.inssec.info/CxWebClient/ViewerMain.aspx?scanid=1018964&projectid=20)
#### Checkmarx Scan Summary
Severity|Count
---|---
High|0
Medium|0
Low|0
Informational|0

#### Violation Summary
Severity|Count
---|---

File: apps/flutter_parent/lib/l10n/app_localizations.dart [hunk 2]
```diff
@@ -682,5 +682,11 @@ class AppLocalizations {
         desc: "Formatted string for when too many items are being notified in a badge, generally something like '99+'",
       );
 
+  String get errorLoadingAnnouncement => Intl.message('There was an error loading this announcement',
+      desc: 'Message shown when an announcement detail screen fails to load');
+
+  String get institutionAnnouncementTitle => Intl.message('Institution Announcement',
+      desc: 'Title text show for institution level announcements');
```
Comment (StephenBrough) [inline apps/flutter_parent/lib/l10n/app_localizations.dart]: show -> shown

File: apps/flutter_parent/assets/html_wrapper.html [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+
+        a {
+            word-wrap: break-word;
+        }
+
+
+</style>
+<body>
+<div id="content">
+    {FLUTTER_CONTENTS}
```
Comment (StephenBrough) [inline apps/flutter_parent/assets/html_wrapper.html]: Not sure if `FLUTTER_CONTENTS` makes sense here since it's not actually Flutter content.

Comment (Blackchai) [reply apps/flutter_parent/assets/html_wrapper.html]: Yeah, this is part of what I'll be cleaning up with the webview PR.

File: apps/flutter_parent/lib/l10n/app_localizations.dart (hunk 2 above)
Comment (Blackchai) [reply apps/flutter_parent/lib/l10n/app_localizations.dart]: :+1:

**Comment by instructure-cx** [comment]
Scan submitted to Checkmarx

**Comment by instructure-cx** [comment]
Scan submitted to Checkmarx

**Comment by instructure-cx** [comment]
### Checkmarx scan completed
[Full Scan Details](https://cxmgr.inssec.info/CxWebClient/ViewerMain.aspx?scanid=1019276&projectid=20)
#### Checkmarx Scan Summary
Severity|Count
---|---
High|0
Medium|0
Low|0
Informational|0

#### Violation Summary
Severity|Count
---|---

**Comment by instructure-cx** [comment]
### Checkmarx scan completed
[Full Scan Details](https://cxmgr.inssec.info/CxWebClient/ViewerMain.aspx?scanid=1019277&projectid=20)
#### Checkmarx Scan Summary
Severity|Count
---|---
High|0
Medium|0
Low|0
Informational|0

#### Violation Summary
Severity|Count
---|---

File: apps/flutter_parent/lib/models/account_notification.dart [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+/// flutter packages pub run build_runner build --delete-conflicting-outputs
+abstract class AccountNotification implements Built<AccountNotification, AccountNotificationBuilder> {
+  @BuiltValueSerializer(serializeNulls: true) // Add this line to get nulls to serialize when we convert to JSON
+  static Serializer<AccountNotification> get serializer => _$accountNotificationSerializer;
+
+  AccountNotification._();
+
+  factory AccountNotification([void Function(AccountNotificationBuilder) updates]) = _$AccountNotification;
+
+  String get id;
```
Comment (StephenBrough) [inline apps/flutter_parent/lib/models/account_notification.dart]: Are all these fields guaranteed to be non-null?

File: apps/flutter_parent/lib/models/announcement.dart [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+/// flutter packages pub run build_runner build --delete-conflicting-outputs
+abstract class Announcement implements Built<Announcement, AnnouncementBuilder> {
+  @BuiltValueSerializer(serializeNulls: true) // Add this line to get nulls to serialize when we convert to JSON
+  static Serializer<Announcement> get serializer => _$announcementSerializer;
+
+  Announcement._();
+
+  factory Announcement([void Function(AnnouncementBuilder) updates]) = _$Announcement;
+
+  String get id;
```
Comment (StephenBrough) [inline apps/flutter_parent/lib/models/announcement.dart]: Also wondering if all these fields are guaranteed to be non-null

File: apps/flutter_parent/lib/models/remote_file.dart [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+/// flutter packages pub run build_runner build --delete-conflicting-outputs
+abstract class RemoteFile implements Built<RemoteFile, RemoteFileBuilder> {
+  @BuiltValueSerializer(serializeNulls: true) // Add this line to get nulls to serialize when we convert to JSON
+  static Serializer<RemoteFile> get serializer => _$remoteFileSerializer;
+
+  RemoteFile._();
+
+  factory RemoteFile([void Function(RemoteFileBuilder) updates]) = _$RemoteFile;
+
+  String get id;
```
Comment (StephenBrough) [inline apps/flutter_parent/lib/models/remote_file.dart]: Same comment that I've got in the other models, hah!

**Pull request merged**